app/scripts/directives/oscPersistentVolumeClaim.js

```diff
@@ -26,7 +27,26 @@ angular.module("openshiftConsole")
         }];
         scope.claim.selectedLabels = [];
         DataService.list({group: 'storage.k8s.io', resource: 'storageclasses'}, {}, function(storageClasses) {
-          scope.storageClasses = storageClasses.by('metadata.name');
+          scope.storageClasses = _.map(storageClasses.by('metadata.name'));
+          var defaultClassAnnotation = "storageclass.beta.kubernetes.io/is-default-class";
+          _.each(scope.storageClasses, function(sclass){
+            if ("annotations" in sclass.metadata && defaultClassAnnotation in sclass.metadata.annotations && sclass.metadata.annotations[defaultClassAnnotation] === "true") {
+              scope.defaultStorageClass = sclass.metadata.name;
+            }
+          });
```

Comment:
This can be

```js
var annotation = $filter('annotation');
scope.defaultStorageClass = _.find(scope.storageClasses, function(storageClass) {
  return annotation(storageClass, 'storageclass.beta.kubernetes.io/is-default-class') === 'true';
});
```

You'll need to add dependency `$filter` above when declaring the the directive.


Even better to add a `isDefaultStorageClass` method to `StorageService`, though. Then this becomes

```js
scope.defaultStorageClass = _.find(scope.storageClasses, StorageService.isDefaultStorageClass);
```

Note that this sets it to the object rather than the name.